files/image_config/hostcfgd/hostcfgd

```diff
@@ -219,6 +220,442 @@ class AaaCfg(object):
         with open(NSS_TACPLUS_CONF, 'w') as f:
             f.write(nss_tacplus_conf)
 
+#Management interface and Management VRF config handling
+
+MGMT_INTF_ADMIN_STATUS_DEFAULT = "up"
+CFG_MGMT_PORT_TABLE = "MGMT_PORT"
+CFG_MGMT_INTF_TABLE = "MGMT_INTERFACE"
+APP_MGMT_PORT_TABLE = "MGMT_PORT_TABLE"
+APP_MGMT_INTF_TABLE = "MGMT_INTF_TABLE"
+CFG_MGMT_VRF      = "MGMT_VRF_CONFIG"
+CFG_MGMT_VRF_KEY  = "vrf_global"
+MGMT_VRF_TABLE_ID = 5000
+
+class MgmtVrfCfg:
+    def __init__(self, config_db):
+        syslog.syslog(syslog.LOG_INFO, 'Initializing MGMT VRF handler')
+        self.config_db = config_db
+        self.mgmt_vrf_table_created = False
+
+    def create_mgmt_vrf_table(self):
+        if self.mgmt_vrf_table_created == False:
+            syslog.syslog(syslog.LOG_INFO, "Create MGMT VRF table:")
+            cmd = 'ip link add name mgmt type vrf table {}'.format(MGMT_VRF_TABLE_ID)
+            if run_command(cmd) == True:
+                self.mgmt_vrf_table_created = True
+
+    def delete_mgmt_vrf_table(self):
+        if self.mgmt_vrf_table_created == True:
+            syslog.syslog(syslog.LOG_INFO, "Delete MGMT VRF table:")
+            cmd = 'ip link delete mgmt'
+            if run_command(cmd) == True:
+                self.mgmt_vrf_table_created = False
+
+    def configure_mgmt_vrf(self):
+        syslog.syslog(syslog.LOG_INFO, "Configure MGMT VRF:")
+
+        self.create_mgmt_vrf_table()
+
+        cmd = 'ip link set dev mgmt up'
+        if run_command(cmd) == False:
+            return
+
+        syslog.syslog(syslog.LOG_INFO, "Add lo intf:")
+        cmd = 'ip link add lo-m type dummy'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip addr add 127.0.0.1/8 dev lo-m'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip link set lo-m up'
+        if run_command(cmd) == False:
+            return
+
+        cmd = 'ip link set dev lo-m master mgmt'
+        if run_command(cmd) == False:
+            return
+
+        self.bind_mgmt_intf()
+
+    def bind_mgmt_intf(self):
```

Comment:
This function is doing two things - binding mgmt interface to master and adding default route to gw which is unrelated to binding. Please revisit